Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -106,7 +110,48 @@ buildRideListItem QRide.RideItem {..} = do
       }
 
 ---------------------------------------------------------------------------------------------------
+ticketRideList :: ShortId DM.Merchant -> Maybe (ShortId Common.Ride) -> Maybe Text -> Maybe Text -> Maybe Text -> Flow Common.TicketRideListRes
+ticketRideList merchantShortId mbRideShortId countryCode mbPhoneNumber _ = do
+  merchant <- findMerchantByShortId merchantShortId
+  let totalRides = 5
+  let mbShortId = coerce @(ShortId Common.Ride) @(ShortId DRide.Ride) <$> mbRideShortId
+  case mbPhoneNumber of
+    Just number -> do
+      no <- getDbHash number
+      let code = fromMaybe "+91" countryCode
+      person <- runInReplica $ QPerson.findByMobileNumberAndMerchant code no merchant.id >>= fromMaybeM (PersonWithPhoneNotFound number)
+      allRides <- QRide.findAllRidesByDriverId person.id
+      let sortedRides = sortBy (compare `on` (.createdAt)) allRides
+      let lastNRides = drop (length sortedRides - totalRides) sortedRides
```

Comment:
What is this logic?

Reply:
we want last 5 rides of the person so sorting on the basis of created at

Reply:
Same, we can move this to query then this handling wont be required